apex/contrib/multihead_attn/self_multihead_attn.py

```diff
@@ -67,10 +88,20 @@ def __init__(self, embed_dim, num_heads, dropout=0., bias=False, include_norm_ad
             else :                   assert False, "Unsupported impl: {} !".format(impl)
 
     def reset_parameters(self):
-        nn.init.xavier_uniform_(self.in_proj_weight)
+        if self.separate_qkv_params:
```

Comment:
Is the separate_qkv_params just for initialization purposes?  I see it concated but never addressed separated, I think.

Reply:
Separation is handled by autograd directly.